tests/python/relax/test_op_manipulate.py

```diff
@@ -2190,5 +2188,299 @@ def test_broadcast_to_infer_struct_info_wrong_input_type():
+    b = tir.Var("b", "int64")
+    x0 = relax.Var("x", R.Tensor((3, 4, a), "float32"))
+    y0 = relax.Var("y", R.Tensor((4, a), "float32"))
+    x1 = relax.Var("x", R.Tensor((3, 4, b + a), "float32"))
+    y1 = relax.Var("x", R.Tensor((1, a + b), "float32"))
+
+    _check_inference(
+        bb, relax.op.collapse_sum_like(x0, y0), relax.TensorStructInfo((4, a), "float32")
+    )
+    _check_inference(
+        bb, relax.op.collapse_sum_like(x1, y1), relax.TensorStructInfo((1, a + b), "float32")
+    )
+
+
+def test_collapse_sum_like_infer_struct_info_shape_var():
+    bb = relax.BlockBuilder()
+    s0 = relax.Var("s", relax.ShapeStructInfo((2, 3, 4)))
+    s1 = relax.Var("s", relax.ShapeStructInfo(ndim=3))
+    s2 = relax.Var("s", relax.ShapeStructInfo())
+    s3 = relax.Var("s", relax.ShapeStructInfo((3, 4)))
+    s4 = relax.Var("s", relax.ShapeStructInfo(ndim=2))
+    s5 = relax.Var("s", relax.ShapeStructInfo())
+    x0 = relax.Var("x", relax.TensorStructInfo(s0, "float32"))
+    x1 = relax.Var("x", relax.TensorStructInfo(s1, "float32"))
+    x2 = relax.Var("x", relax.TensorStructInfo(s2, "float32"))
+    y0 = relax.Var("y", relax.TensorStructInfo(s3, "float32"))
+    y1 = relax.Var("y", relax.TensorStructInfo(s4, "float32"))
+    y2 = relax.Var("y", relax.TensorStructInfo(s5, "float32"))
+
+    _check_inference(bb, relax.op.collapse_sum_like(x0, y0), relax.TensorStructInfo(s3, "float32"))
+    _check_inference(bb, relax.op.collapse_sum_like(x1, y1), relax.TensorStructInfo(s4, "float32"))
+    _check_inference(bb, relax.op.collapse_sum_like(x2, y2), relax.TensorStructInfo(s5, "float32"))
+
+
+def test_collapse_sum_like_infer_struct_info_more_input_dtype():
+    bb = relax.BlockBuilder()
+    x0 = relax.Var("x", R.Tensor((2, 3, 4), "float16"))
+    x1 = relax.Var("x", R.Tensor((2, 3, 4), "int8"))
+    y0 = relax.Var("y", R.Tensor((3, 4), "float16"))
+    y1 = relax.Var("y", R.Tensor((3, 4), "int8"))
+
+    _check_inference(
+        bb, relax.op.collapse_sum_like(x0, y0), relax.TensorStructInfo((3, 4), "float16")
+    )
+    _check_inference(bb, relax.op.collapse_sum_like(x1, y1), relax.TensorStructInfo((3, 4), "int8"))
+
+
+def test_collapse_sum_like_wrong_input_type():
+    bb = relax.BlockBuilder()
+    x0 = relax.Var("x", R.Tensor((3, 4, 5), "float32"))
+    x1 = relax.Var("x", relax.ShapeStructInfo((4, 5)))
+    x2 = relax.Var("x", relax.FuncStructInfo([], R.Tensor((2, 3, 4), "float32")))
+
+    with pytest.raises(TVMError):
+        bb.normalize(relax.op.collapse_sum_like(x0, x1))
+
+    with pytest.raises(TVMError):
+        bb.normalize(relax.op.collapse_sum_like(x2, x0))
+
+
+def test_collapse_sum_like_check_shape_failure():
+    bb = relax.BlockBuilder()
+    x = relax.Var("x", R.Tensor((3, 4, 5), "float32"))
+    y = relax.Var("y", R.Tensor((3, 6, 5), "float32"))
+    a = tir.Var("a", "int64")
+    b = tir.Var("b", "int64")
+    z = relax.Var("z", R.Tensor([a], "float32"))
+    w = relax.Var("w", R.Tensor([b], "float32"))
```

Comment:
Also test the cases like
```python
sx = relax.Var("sx", relax.ShapeStructInfo((3, 4, 5)))
sy = relax.Var("sy", relax.ShapeStructInfo((3, 6, 5)))
x = relax.Var("x", relax.TensorStructInfo(sx, "float32"))
y = relax.Var("y", relax.TensorStructInfo(sy, "float32"))
```
and similar symbolic cases. Please add such tests for `collapse_sum_to` as well. Thanks!